components/tools/OmeroPy/test/integration/clitest/test_duplicate.py

```diff
@@ -128,3 +150,55 @@ def testDuplicateSingleObjectDryRunReport(self, capfd):
         assert '%s:%s' % (object_type, objs[0].id.val) in out
         # Check object has been found in two different places of the output
         assert out.find(obj_arg) != out.rfind(obj_arg)
+
+    def testBasicHierarchyDuplication(self, capfd):
+        namep = self.uuid()
+        named = self.uuid()
+        namei = self.uuid()
+        proj = self.make_project(namep)
+        dset = self.make_dataset(named)
+        img = self.make_image(namei)
+
+        self.link(proj, dset)
+        self.link(dset, img)
+
+        self.args += ['Project:%s' % proj.id.val]
+        self.duplicate(capfd)
+
+        pp = omero.sys.ParametersI()
+        pp.addString("name", namep)
+        query = "select obj from Project obj where obj.name=:name"
+        objsp = self.query.findAllByQuery(query, pp)
+
+        pd = omero.sys.ParametersI()
+        pd.addString("name", named)
+        query = "select obj from Dataset obj where obj.name=:name"
+        objsd = self.query.findAllByQuery(query, pd)
```

Comment:
Then here check (a) there are two datasets and (b) the new dataset is linked to the new project.